src/main/java/org/opentripplanner/transit/service/TransitModel.java

```diff
@@ -222,6 +222,10 @@ public void updateCalendarServiceData(
    */
   @Nullable
   public FeedScopedId getOrCreateServiceIdForDate(LocalDate serviceDate) {
+    if (serviceDate == null) {
+      return null;
+    }
+
```

Comment:
Move this chek out of the method, to where it is called from

Reply:
I have moved this to the added trip method and added a Update result error for this case.

I also added a @nonnull annotation on the method per leonards recommendation